pkg/apis/core/types.go

```diff
@@ -3430,6 +3440,19 @@ type ServiceSpec struct {
 	// of peer discovery.
 	// +optional
 	PublishNotReadyAddresses bool
+
+	// The Service IPFamily  specifies whether this Service has a preference for a particular
+	// IP family (e.g. IPv4 vs. IPv6).  If a specific IP family is specified, addresses
+	// of that family will be required for the clusterIP, loadBalancerIP,
+	// loadBalancerSourceRanges, externalIPs, and ports[*].nodePort fields as
+	// well as for populating associated endpoints.  If no IP family is specified,
+	// the cluster-wide default will be used for the clusterIP, loadBalancerIP, and
+	// loadBalancerSourceRanges fields, the externalIPs field and associated endpoints,
+	// and node ports will be bound to addresses regardless of IP family.  This field is immutable
+	// after creation. Assigning a ServiceIPFamily not available in the cluster (e.g. IPv6 in IPv4
+	// only cluster) is an error condition and will fail during ClusterIP assignment.
+	// +optional
```

Comment:
it's not optional, it's required with a default.  We need to be clearer about that.